pkgs/build-support/docker/nix-prefetch-docker

```diff
@@ -121,7 +121,7 @@ fi
 
 sourceUrl="docker://$imageName@$imageDigest"
 
-tmpPath="$(mktemp -d "${TMPDIR:-/tmp}/skopeo-copy-tmp-XXXXXXXX")"
+tmpPath="$(mktemp -d "$(realpath "${TMPDIR:-/tmp}/skopeo-copy-tmp-XXXXXXXX")")"
```

Comment:
How about:
```suggestion
tmpPath="$(realpath "$(mktemp -d --tmpdir skopeo-copy-tmp-XXXXXXXX)")"
```
and analoguously for all the others?

Reply:
This actually used to use --tmpdir over 10 years ago, but was changed to use TMPDIR in https://github.com/NixOS/nixpkgs/commit/5e7a1cf955e43e1fa3c1c157c0d6961e1bcf9801.

It does look like my /usr/bin/mktemp does support --tmpdir on macOS 15.1.1, but I'm not sure how far back that goes. I could add coreutils to all the prefetch scripts to try & make sure we've always got a usable `mktemp --tempdir` ..?

Reply:
Ah, I knew I kept my trusty 10.12 VM around for something:

![Screenshot of macOS 10.12 confirming that its `mktemp` doesn’t support `--tmpdir`.](https://github.com/user-attachments/assets/237df9f4-aaed-4974-a7ad-064a2a472b32)

So this would be fine as‐is for 25.05 I think but not for 24.11. However, yes, we probably ought to just add `coreutils` to the list of common dependencies in `pkgs/tools/package-management/nix-prefetch-scripts/default.nix`; all our other Bash scripts assume GNU coreutils anyway.